numpy/lib/npyio.py

```diff
@@ -686,31 +684,44 @@ def _savez(file, args, kwds, compress, allow_pickle=True, pickle_kwargs=None):
 
     zipf = zipfile_factory(file, mode="w", compression=compression)
 
-    # Stage arrays in a temporary file on disk, before writing to zip.
-
-    # Since target file might be big enough to exceed capacity of a global
-    # temporary directory, create temp file side-by-side with the target file.
-    file_dir, file_prefix = os.path.split(file) if _is_string_like(file) else (None, 'tmp')
-    fd, tmpfile = tempfile.mkstemp(prefix=file_prefix, dir=file_dir, suffix='-numpy.npy')
-    os.close(fd)
-    try:
+    if sys.version_info >= (3, 6):
+        # Since Python 3.6 it is possible to write directly to a ZIP file.
         for key, val in namedict.items():
             fname = key + '.npy'
-            fid = open(tmpfile, 'wb')
-            try:
-                format.write_array(fid, np.asanyarray(val),
+            val = np.asanyarray(val)
+            force_zip64 = val.nbytes >= 2**30
+            with zipf.open(fname, 'w', force_zip64=force_zip64) as fid:
```

Comment:
Is `force_zip64` an unrelated fix here?

Reply:
and should this be `wb` to match the one below?

Reply:
> The mode parameter, if included, must be 'r' (the default) or 'w'

I suspect the zip files are automatically binary.

Reply:
`force_zip64` is needed when write a stream. Tests are failed without this option.

`'r'` and `'w'` are the only supported options. Opened file-like objects are binary.

Reply:
Looks like zip64 has been available since 2.7, but the `force_zip64` keyword is new. It is used for files of unknown size that may exceed 2 GiB. As we are formatting the array, I suppose it qualifies as of unknown size, but probably not by much.

Reply:
Should we be using `force_zip64` on all version of python 3, not just 3.6?

Reply:
`force_zip64` doesn't exist in earlier versions.